pkg/controller/jobs/statefulset/statefulset_webhook.go

```diff
@@ -64,23 +69,34 @@ func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
 	log.V(5).Info("Applying defaults")
 
 	queueName := jobframework.QueueNameForObject(ss.Object())
-	if queueName == "" {
-		return nil
-	}
 
-	if ss.Spec.Template.Labels == nil {
-		ss.Spec.Template.Labels = make(map[string]string, 2)
+	// Do not manage a statefulset without a queue name unless the namespace selector also matches
+	if features.Enabled(features.ManagedJobsNamespaceSelector) && wh.manageJobsWithoutQueueName && queueName == "" {
```

Comment:
I think this is a bit hard to follow, espacially because `manageJobsWithoutQueueName` and `queueName` are checked in both `if`s - here and below. The code is also duplicated with Jobs.

So, I would suggest to commonize the code, maybe like that:
```golang
func WorkloadManagedByKueue(ctx context.Context, jobObj client.Object, k8sClient client.Client,
	manageJobsWithoutQueueName bool, managedJobsNamespaceSelector labels.Selector) (bool, error) {
	// If the owner is managed, then the workload is also managed
	if owner := metav1.GetControllerOf(jobObj); owner != nil && IsOwnerManagedByKueue(owner) {
	   return true, nil
	}
	// we manage Jobs with queue name
	if QueueNameForObject(jobObj) != "" { 
	     return true, nil
	}
	// Manage Jobs if manageJobsWithoutQueueName and namespace selector matches
	if features.Enabled(features.ManagedJobsNamespaceSelector) && manageJobsWithoutQueueName {
		ns := corev1.Namespace{}
		err := k8sClient.Get(ctx, client.ObjectKey{Name: jobObj.GetNamespace()}, &ns)
		if err != nil {
			return false, fmt.Errorf("failed to get namespace: %w", err)
		}
		if managedJobsNamespaceSelector.Matches(labels.Set(ns.GetLabels())) {
			return true, nil
		}
	}
	return false, nil
}
```
I know this is more complex for pods, so we could commonize with pods at another chance. If there is more work than on the surface I'm ok with a follow up.

Reply:
Good suggestion.  It was straightforward to pull out helper method and make it easier to see what is happening.